### PR TITLE
Add support for Rubicson pool thermometer 48942

### PR DESCRIPTION
**Pull request by robho**

### CODE REVIEW

File: src/devices/rubicson_pool_48942.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    Copyright (C) 2022 Robert Högberg
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+/**
+   The device uses OOK and fixed period PWM.
```
Comment (zuckschwerdt) [inline src/devices/rubicson_pool_48942.c:13]: Don't indent this block (i.e. remove two spaces)

File: src/devices/rubicson_pool_48942.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+   - 0 is encoded as 240 us pulse and 480 us gap,
+   - 1 is encoded as 480 us pulse and 240 us gap.
+
+   A transmission consists of an initial preamble followed by sync
+   pulses and the data. Sync pulses and data are sent twice.
+
+   Preamble:
+    __      ____      ____      ____      ____
```
Comment (zuckschwerdt) [inline src/devices/rubicson_pool_48942.c:21]: Indent verbatim sections with exactly four spaces

File: src/devices/rubicson_pool_48942.c [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    - X: CRC
+*/
+
+#include "decoder.h"
+
+static int rubicson_pool_48942_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    int row = bitbuffer_find_repeated_row(bitbuffer, 2, 41);
+    if (row < 0 || bitbuffer->bits_per_row[row] != 41)
+      return DECODE_ABORT_LENGTH;
```
Comment (zuckschwerdt) [inline src/devices/rubicson_pool_48942.c:54]: Indent four spaces each level.

File: src/devices/rubicson_pool_48942.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    int channel = (b[0] >> 4) + 1;
+    int random_id = (b[0] & 0x0F) << 6 | (b[1] & 0xFC) >> 2;
+    int battery_low = b[2] >> 7;
+    float temp_c = ((((b[2] & 0x7F) << 4) | (b[3] >> 4)) - 1024) * 0.1f;
+
+    decoder_log_bitbuffer(decoder, 1, __func__, bitbuffer, "");
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",          "",             DATA_STRING,  "Rubicson Pool Thermometer 48942",
```
Comment (zuckschwerdt) [inline src/devices/rubicson_pool_48942.c:75]: "model" needs to be "Rubicson-48942"

**Comment by robho** [comment]
I hope I got the indentation correct this time. Please take a look.

File: src/devices/rubicson_pool_48942.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    Copyright (C) 2022 Robert Högberg
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+/**
+The device uses OOK and fixed period PWM.
```
Comment (zuckschwerdt) [inline src/devices/rubicson_pool_48942.c:13]: Line 2+3 (name and empty line) should be repeated here (without indent).

**Comment by zuckschwerdt** [comment]
First line for the doc comment needs to be a name, otherwise this looks good. Thanks!

File: src/devices/rubicson_pool_48942.c [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+Preamble:
+     __      ____      ____      ____      ____
+    |  |____|    |____|    |____|    |____|    |__________
+    480 980  980  980  980  980  980  980  980  3880     [us]
+
+Sync pulses:
+     ___     ___     ___     ___
+    |   |___|   |___|   |___|   |___
+     730 730 730 730 730 730 730 730    [us]
```
Comment (robho) [inline src/devices/rubicson_pool_48942.c:28]: Is it important that this line is indented by 4 also? Imo it looks better together with the pulses above with 5 character indentation.

Is this markdown rendered somewhere?

Comment (zuckschwerdt) [reply src/devices/rubicson_pool_48942.c:30]: > Is it important that this line is indented by 4 also? Imo it looks better together with the pulses above with 5 character indentation.

Yes, four indents for a preformatted block. All indents are removed.

> Is this markdown rendered somewhere?

Yes ;) https://triq.org/explorer/

Comment (robho) [reply src/devices/rubicson_pool_48942.c]: Ok, I changed indentation for this line in the latest version.

**Pull request merged**